tests/invalid_models_tests/test_models.py

```diff
@@ -1325,3 +1325,38 @@ class Meta:
                 constraints = [models.CheckConstraint(check=models.Q(age__gte=18), name='is_adult')]
 
         self.assertEqual(Model.check(databases=self.databases), [])
+
+    def test_check_unique_conditions(self):
+        class Model(models.Model):
+            _id = models.IntegerField()
+
+            class Meta:
+                constraints = [
+                    models.UniqueConstraint(name='id_gt_10', fields=['_id'], condition=models.Q(_id__gte=10))
+                ]
+
+        errors = Model.check(databases=self.databases)
+        warn = Warning(
+            "%s does not support partial indexing, so the unique constraint "
+            "won't be created." % connection.display_name,
+            hint=(
+                "A constraint won't be created. Silence this warning if you "
+                "don't care about it."
+            ),
+            obj=Model,
+            id='models.W027',
+        )
+        expected = [] if connection.features.supports_partial_indexes else [warn]
+        self.assertCountEqual(errors, expected)
```

Comment:
This should use `assertEqual` and not `assertCountEqual`, otherwise there is no point in entirely recreating the expected warning, you could just hardcode 1 or 0.